File: member-service/src/main/java/com/hedvig/memberservice/events/MemberCreatedEvent.java
package com.hedvig.memberservice.events;

import com.hedvig.memberservice.aggregates.MemberStatus;
import lombok.Value;

import java.time.Instant;

@Value
public class MemberCreatedEvent {

    public Long id;

    public MemberStatus status;

    public Instant createdOn;

    public MemberCreatedEvent(Long id, MemberStatus status) {
        this.id = id;
        this.status = status;
        this.createdOn = Instant.now();
    }
}

File: member-service/src/main/java/com/hedvig/memberservice/events/MemberCreatedEvent.kt
package com.hedvig.memberservice.events

import com.hedvig.memberservice.aggregates.MemberStatus
import java.time.Instant

class MemberCreatedEvent(
    val id: Long,
    val status: MemberStatus,
    val createdOn: Instant = Instant.now()
) {
    constructor(
        id: Long,
        status: MemberStatus
    ) : this(id, status, Instant.now())
}
